### PR TITLE
Shopify common js

### LINKED ISSUE #20: Add shopify_common.js module
### Problem

Starter theme uses `Shopify.CountryProvinceSelector()` and `Shopify.postLink()` in [customers/addresses.js](https://github.com/Shopify/starter-theme/blob/master/src/assets/scripts/templates/customers/addresses.js#L15-L24) but `shopify_common.js` is not loaded and the following error is produced

screen shot 2018-06-19 at 6 45 03 pm

### shopify_common.js

Contains functions to update the contents of the Province `<select>` as you select different values in the Country `<select>`

### Adding it as another package

If we had this functions in another package in theme-scripts we could use it as dependency in starter-theme

_package.json_
```diff
  "dependencies": {
+   "@shopify/theme-common": "^1.0.0-alpha.3",
    "@shopify/theme-a11y": "^1.0.0-alpha.3",
    "@shopify/theme-cart": "^1.0.0-alpha.3",
    "@shopify/theme-currency": "^1.0.0-alpha.3",
```

_src/assets/scripts/templates/customers/addresses.js_
```diff
 import $ from 'jquery';
+import { CountryProvinceSelector, postLink } from '@shopify/theme-common';

-new Shopify.CountryProvinceSelector(countrySelector, provinceSelector, {
+new CountryProvinceSelector(countrySelector, provinceSelector, {

-Shopify.postLink(`/account/addresses/${formId}`, {
+postLink(`/account/addresses/${formId}`, {
```

### PR DESCRIPTION
**Pull request by montalvomiguelo**
Adds theme-addresses package to fix #20 

We can use this library instead of loading `{{ 'shopify_common.js' | shopify_asset_url | script_tag }}` in Shopify/starter-theme to fix Shopify/starter-theme#74

### CODE REVIEW

File: packages/theme-addresses/addresses.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    if (value === option.value || value === option.innerHTML) {
+      selector.selectedIndex = i;
+      return i;
+    }
+  }
+  return -1;
+}
+
+// send request as a POST
+export function postLink(path, options = {}) {
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/addresses.js]: Not sure if we can use Fetch here... response of request _POST, DELETE_ to `/account/addresses/` is always a redirect so after calling `postLink` we always get back to `/account/addresses/` with updated addresses

https://cdn.shopify.com/s/shopify/shopify_common.js

File: packages/theme-addresses/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+```
+yarn add @shopify/theme-addresses
+```
+
+and then import the functions you wish to use through ES6 imports:
+
+```
+import * as addresses from '@shopify/theme-addresses`;
+```
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/README.md]: Added links to builds

File: packages/theme-addresses/addresses.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
+/*
+ * Shopify Addresses
+ * ------------------------------------------------------------------------------
+ * Contains useful functions that can be used in addresses.liquid
+ */
+
+// set a given selector with value, if value is one of the options
+export function setSelectorByValue(selector, value) {
+  for (let i = 0, count = selector.options.length; i < count; i++) {
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/addresses.js]: Now using es5

File: packages/theme-addresses/addresses.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
+/*
+ * Shopify Addresses
+ * ------------------------------------------------------------------------------
+ * Contains useful functions that can be used in addresses.liquid
+ */
+
+// set a given selector with value, if value is one of the options
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/addresses.js]: It does not return any value now

File: packages/theme-addresses/package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  "author": "Shopify Inc",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/Shopify/theme-scripts/issues"
+  },
+  "homepage": "https://github.com/Shopify/theme-scripts#readme",
+  "scripts": {
+    "prepublish": "npm run build",
+    "build": "npm run build:es5 & npm run build:cjs",
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/package.json]: Updated build scripts

File: packages/theme-addresses/README.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# @shopify/theme-addresses
+
+## Getting Started
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/README.md]: It looks like the Readme file of _theme-a11y_ should I add something else?

File: packages/theme-addresses/addresses.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+ *   <select id="country_selector" name="country" data-default="Canada">
+ *     <option data-provinces="[['Alberta','Alberta'],['Ontario','Ontario'],['British Columbia','British Columbia'],...] value="Canada">Canada</option>
+ *     ...
+ *   </select>
+ *   <select id="province_selector" name="province" data-default="Ontario">
+ *     <option value="Ontario">Ontario</option>
+ *     ...
+ *   </select>
+ */
+export class CountryProvinceSelector {
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/addresses.js]: es5 now

File: packages/theme-addresses/package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "@shopify/theme-addresses",
+  "version": "1.0.0-alpha.7",
+  "description": "A library that contains useful functions that can be used in addresses.liquid",
+  "main": "dist/addresses.cjs.js",
+  "moules": "dist/addresses.es5.js",
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/package.json]: module

File: packages/theme-addresses/addresses.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (t-kelly) [reply packages/theme-addresses/addresses.js]: Do we need it though? Is it being used?

[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/addresses.js]: We call this when we want to delete an address 

https://github.com/Shopify/starter-theme/blob/master/src/assets/scripts/templates/customers/addresses.js#L57-L59

**Comment by gabyhernan** [comment]
Hi there, currently facing this same issue. Would love to contribue as well if you can could add more support docs in place.

File: packages/theme-addresses/package.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@shopify/theme-addresses",
+  "version": "1.0.0-alpha.7",
+  "description": "A library that contains useful functions that can be used in addresses.liquid",
+  "main": "dist/theme-addresses.cjs.js",
+  "moule": "theme-addresses.js",
```
[the remark this answers is not in the document]
Comment (montalvomiguelo) [reply packages/theme-addresses/package.json]: omg sry

**Pull request closed without merging**